packages/front-end/components/Settings/WebhooksModal.tsx

```diff
@@ -56,14 +89,92 @@ const WebhooksModal: FC<{
       value: "",
     });
   }
+  const SDKFilterFields = () => (
+    <>
+      <Toggle
+        id="sendPayload"
+        value={!!form.watch("sendPayload")}
+        setValue={(value) => {
+          console.log(value);
+          console.log(!!form.watch("sendPayload"));
+          form.setValue("sendPayload", value);
+        }}
+      />
+      <label htmlFor="sendPayload">Send Payload</label>
+      <SelectField
+        label="Method"
+        required
+        placeholder="POST"
+        value={form.watch("httpMethod")}
+        onChange={(httpMethod: WebhookMethod) =>
+          form.setValue("httpMethod", httpMethod)
+        }
+        options={methodTypes.map((e) => ({ label: e, value: e }))}
+      />
+      {headerJsonEditor()}
+    </>
+  );
+  const validateHeaders = (headers: string) => {
+    try {
+      JSON.parse(headers);
+      setValidHeaders(true);
+    } catch (error) {
+      setValidHeaders(false);
+    }
+  };
+  const headerJsonEditor = () => (
+    <CodeTextArea
+      label="Headers"
+      language="json"
+      value={form.watch("headers")}
+      setValue={(headers) => {
+        validateHeaders(headers);
+        form.setValue("headers", headers);
+      }}
+      helpText={
+        <>
+          {!validHeaders ? (
+            <div className="alert alert-danger mr-auto">Invalid JSON</div>
+          ) : (
+            <div>JSON format for headers.</div>
+          )}
+        </>
+      }
+    />
+  );
+  const nonSDKFilterFields = () => (
+    <>
+      <Field
```

Comment:
consider <SelectField> instead

Reply:
yeah i might since im touching that file and would be good to get the old up to the newer stuff